internal/nexodus/nexodus.go

```diff
@@ -271,12 +278,28 @@ func (ax *Nexodus) Start(ctx context.Context, wg *sync.WaitGroup) error {
 		return fmt.Errorf("handleKeys: %w", err)
 	}
 
-	user, _, err := ax.client.UsersApi.GetUser(ctx, "me").Execute()
+	var user *public.ModelsUser
+	err = util.RetryOperation(ctx, 10*time.Second, 3, func() error {
+		user, _, err = ax.client.UsersApi.GetUser(ctx, "me").Execute()
+		if err != nil {
+			ax.logger.Debugf("get user error - retrying: %v", err)
+			return err
+		}
+		return nil
+	})
 	if err != nil {
 		return fmt.Errorf("get user error: %w", err)
 	}
 
-	organizations, _, err := ax.client.OrganizationsApi.ListOrganizations(ctx).Execute()
+	var organizations []public.ModelsOrganization
+	err = util.RetryOperation(ctx, 10*time.Second, 3, func() error {
+		organizations, _, err = ax.client.OrganizationsApi.ListOrganizations(ctx).Execute()
+		if err != nil {
```

Comment:
If ListOrganizations can throw specific error, that can help enhance this logic even better. With specific error, we can just retry on specific event, and skip on other errors like 404.

Reply:
Pretty much any error here will break things (or any of the initial API calls up to reconcile(false) . If we got a 404 here we would want to exit as well right?

Reply:
If you put the controller down, it should fail with different error message? But anyways, this comment was not in the context of this PR, but moreover to further improve the logic in future, whenever we improve the client apis.